app/Policies/MacAddressPolicy.php

```diff
@@ -16,57 +15,31 @@ public function before(User $user)
         if ($user->isAdmin()) {
             return true;
         }
-        if (!$user->hasRole(Role::INTERNET_USER)) {
-            return false;
-        }
-    }
-
-    /**
-     * Determine whether the user can view any mac addresses (in general).
-     *
-     * @param User $user
-     * @return mixed
-     */
-    public function viewAny(User $user): bool
-    {
-        return false;
-    }
-
-    /**
-     * Determine whether the user can view the mac address.
-     *
-     * @param User $user
-     * @param MacAddress $macAddress
-     * @return mixed
-     */
-    public function view(User $user, MacAddress $macAddress): bool
-    {
-        return $user->id === $macAddress->user_id;
     }
 
     /**
-     * Determine whether the user can create mac addresses.
+     * Determine whether the user can create mac address instances.
      *
      * @param User $user
-     * @return mixed
+     * @return bool
      */
     public function create(User $user): bool
     {
-        return true;
+        return $user->can('handle', $user->internetAccess);
```

Comment:
This only redirects it to the InternetAccessPolicy's handle which only checks currently that it can not create a Mac for other users. Which is true in every case in this call :D I'll just delete it. (The internet access should exist by default, but if not it also returns 403 for null. The expired check is not that necessary, just creates more tension.)